### ISSUE TITLE
Getting wrongly calculated values while using keras.metrics.R2Score()

### ISSUE THREAD

**Issue by jayaBalaR** [keras-team/keras#21065]
Hi
In our deep learning model, we are using keras.metrics.R2Score()  and sending in the call backs list while doing a fit.
But the values returned by the function are not correct. Please can you help with a fix for this code.

Here is the skeletal code where we are using the function. we get highly negative values for r2 score .
def build_model(input_shape, output_width, optimizer='adam'):
    """
    Builds a Sequential model with Conv1D, LSTM, Dropout, and Dense layers.
    Includes regularization and dropout to prevent overfitting.
    """
    print("entered build_model")
    print("input_shape=", input_shape)
    print("output_width=", output_width)
    model = tf.keras.Sequential()

    # Reshape input if necessary: This addresses the potential extra dimension
    # model.add(layers.Reshape(target_shape=(-1, input_shape[-1]), input_shape=input_shape))  # Remove this line
    # target_shape uses -1 to infer batch_size and assumes the last dimension of input_shape is features.

    # 1st Conv1D Layer
    # Adjust input_shape to (timesteps, features)

    # Add Batch Normalization
    

    # 1st LSTM Layer
    

    # 2nd LSTM Layer
    

    # Dropout Layer to prevent overfitting
    

    # Dense Layer with output size matching the number of timesteps in output
    

    # Compile the model
    if optimizer == 'rmsprop':
        optimizer_fn = tf.keras.optimizers.RMSprop()
    elif optimizer == 'adam':
        optimizer_fn = tf.keras.optimizers.Adam()
    elif optimizer == 'sgd':
        optimizer_fn = tf.keras.optimizers.SGD()
    else:
        raise ValueError(f"Unknown optimizer: {optimizer}")

    r2score = keras.metrics.R2Score()

    model.compile(optimizer=optimizer_fn, loss='huber')

    return model

**Comment by mehtamansi29** [comment]
Hi @jayaBalaR -

Thanks for reporting this issue. Here in your code you are not defining r2score metrics while model compilation. 
Attached [gist](https://colab.sandbox.google.com/gist/mehtamansi29/205f33e6c4f1e1094c124b8f9364412e/21065-getting-wrongly-calculated-values-while-using-keras-metrics-r2score.ipynb) shows the code with training model on dummy data and you can see r2_score is negative intial epochs then it decreases and become positive and improve towards to 1.

**Comment by jayaBalaR** [comment]
Hi @mehtamansi29 
Thank you for the resolution.

**Comment by mehtamansi29** [comment]
Hi @jayaBalaR - Please closed the task if your issue is resolved.
